### ISSUE TITLE
[Translation tool] ko

### ISSUE THREAD

**Issue by ukits** [briannesbitt/Carbon#1908]
|English|ko|comments|
|--|--|--|
|2yrs|2 년|2년|
|2mos|2 개월|2개월|
|2d|2 일|2일|
|2h|2 시간|2시간|
|2m|2 분|2분|
|2s|2 초|2초|
|1 second|몇 초|1초|
|2 hours before|2시간 이전|2시간 전|
|2 hours after|2시간 이후|2시간 후|

**Comment by kylekatarnls** [comment]
Thanks for your help @ukits can you tell if `이전` will apply for any interval and unit such as "2 years and 5 months"?

**Comment by ukits** [comment]
In korean, `이전` and `전` are the same. `이후` and `후` are also same. `전`/`후` is more common for many uses I think. `전` is suitable for both of "2 years and 5 months ago" and "2 years and 5 months before".

**Comment by kylekatarnls** [comment]
CLDR confirms your position about digit spacing.

While Google Translate and glibc keep the space:
https://translate.google.com/#view=home&op=translate&sl=en&tl=ko&text=3%20hours%20ago

Is there an official rule? Is it controversial or needs to be enable via an option like we did for Chinese?
```php
echo Carbon::now()->locale('zh')->subMonths(2)->diffForHumans();
echo "\n";
echo Carbon::now()->locale('zh')->subMonths(2)->diffForHumans(['join' => ' ']);
```

This outputs:
```
2个月前
2 个月前
```

**Comment by ukits** [comment]
Nationall Institute of Korean Language(http://kornorms.korean.go.kr/regltn/regltnView.do?regltn_code=0001&regltn_no=182#a182 , see paragraph 43 and the second box on it, in Korean) said,
"Nouns that represent units have spacing in front of. However, they might have no spacing when used in combination with **numbers** or order."

Google translate was not wrong. It is also allowed.

In case of Carbon, because they used with number always, spacing might not be required.
For example, `2 년 5 개월 6 일 15 시간 13 분 45 초` is kind of awkward. `2년 5개월 6일 15시간 13분 45초` is better.

The most of Korean people want no spacing. Nevertheless, official rule allows both of them. 'join' option is good I think. 
But, Chinese 'join' option will shrink all spaces like `15시간13분45초`. `15시간 13분 45초` is expected for "ko" locale.

**Comment by kylekatarnls** [comment]
CLDR is a reference used by many libraries as a standard (Angular for example): http://cldr.unicode.org/

You can download a database as an XML file and it contains relative time translations for a lot of languages.

**Comment by kylekatarnls** [comment]
Finally I simply removed spaces as we had a divergence between short and long format while they should be the same for Korean. Then we want no space by default, so the solution used for Chinese is not applicable.

It will be released in 2.27.
